rosbag2_transport/CMakeLists.txt

```diff
@@ -84,63 +85,72 @@ ament_export_interfaces(export_${PROJECT_NAME})
 ament_export_libraries(${PROJECT_NAME})
 ament_export_dependencies(rosbag2_cpp rosbag2_compression yaml_cpp_vendor)
 
-if(BUILD_TESTING)
-  find_package(ament_cmake_gmock REQUIRED)
-  find_package(ament_index_cpp REQUIRED)
-  find_package(ament_lint_auto REQUIRED)
-  find_package(test_msgs REQUIRED)
-  find_package(rosbag2_test_common REQUIRED)
-  # explicitly disable cppcheck
-  set(ament_cmake_cppcheck_FOUND TRUE)
-  ament_lint_auto_find_test_dependencies()
+function(create_tests_for_rmw_implementation)
+  message(STATUS "Creating tests for '${rmw_implementation}'")
+  set(rmw_implementation_env_var RMW_IMPLEMENTATION=${rmw_implementation})
 
-  ament_add_gmock(test_info
+  set(test_info test_info${target_suffix})
+  ament_add_gmock("${test_info}"
     test/rosbag2_transport/test_info.cpp
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
-  if(TARGET test_info)
-    target_link_libraries(test_info rosbag2_transport)
-    ament_target_dependencies(test_info rosbag2_test_common)
+    ENV "${rmw_implementation_env_var}"
+    WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
+  if(TARGET ${test_info})
+    target_link_libraries(${test_info} rosbag2_transport)
+    ament_target_dependencies(${test_info}
+      ${rmw_implementation} rosbag2_test_common)
   endif()
```

Comment:
Let's extract this logic into a single macro like `declare_rosbag2_test(test_info, test/rosbag2_transport/test_info.cpp)` or something? The current diff is copying and pasting the same logic everywhere.

Reply:
fair enough. done